lib/sinatra/base.rb

```diff
@@ -1416,7 +1416,11 @@ def quit!
         return unless running?
         # Use Thin's hard #stop! if available, otherwise just #stop.
         running_server.respond_to?(:stop!) ? running_server.stop! : running_server.stop
-        $stderr.puts "== Sinatra has ended his set (crowd applauds)" unless handler_name =~/cgi/i
+
+        unless lifecycle_announcements == false || handler_name =~/cgi/i
```

Comment:
What about to use positive condition?

`if lifecycle_announcements || handle_name !~ /cgi/g`

Since that condition is used twice, what about to wrap it into some method?

Reply:
I think the right positive condition is `if lifecycle_announcements && handle_name !~ /cgi/g` ?